modules/math/src/main/java/com/opengamma/strata/math/impl/interpolation/data/ArrayInterpolator1DDataBundle.java

```diff
@@ -85,35 +86,32 @@ private static void swap(final double[] keys, final double[] values, final int f
   }
 
   @Override
-  public boolean containsKey(final Double key) {
-    if (key == null) {
-      return false;
-    }
+  public boolean containsKey(double key) {
     return Arrays.binarySearch(_keys, key) >= 0;
   }
 
   @Override
-  public Double firstKey() {
+  public double firstKey() {
     return _keys[0];
   }
 
   @Override
-  public Double firstValue() {
+  public double firstValue() {
     return _values[0];
   }
 
   @Override
-  public Double get(final Double key) {
-    final int index = Arrays.binarySearch(_keys, key);
+  public double get(double key) {
+    int index = Arrays.binarySearch(_keys, key);
     if (index < 0) {
-      return null;
+      return Double.NaN;
```

Comment:
Do we definitely want to be returning `NaN`? What's the expected behaviour in this case? Is it possible to produce a useful answer after returning `NaN`? If not then maybe it should throw an exception instead. `NaN` has a nasty habit of quietly spreading and infecting things and it can be hard to track down where it came from.